sdk/cosmos/azure-cosmos-tests/src/test/java/com/azure/cosmos/CosmosItemWriteRetriesTest.java

```diff
@@ -0,0 +1,247 @@
+        FaultInjectionRule rule = null;
+
+        if (injectFailure) {
+            rule = injectFailure(container, FaultInjectionOperationType.CREATE_ITEM, suppressServiceRequests);
+        }
+
+        try {
+            String id = UUID.randomUUID().toString();
+            executeAndValidate(() -> container.createItem(
+                getDocumentDefinition(id), new PartitionKey(id), options),
+                id,
+                expectedStatusCode,
+                false);
+        } finally {
+            if (rule != null) {
+                rule.disable();
+                container.getDatabase().getClient().close();
+                logger.info("JSON-Traces: {}", this.mockTracer.toJson());
+            }
+        }
+    }
+
+    private FaultInjectionRule injectFailure(
+        CosmosAsyncContainer container,
+        FaultInjectionOperationType operationType,
+        Boolean suppressServiceRequests) {
+
+        FaultInjectionServerErrorResultBuilder faultInjectionResultBuilder = FaultInjectionResultBuilders
+            .getResultBuilder(FaultInjectionServerErrorType.RESPONSE_DELAY)
+            .delay(Duration.ofMillis(1500))
+            .times(1);
+
+        if (suppressServiceRequests != null) {
+            faultInjectionResultBuilder.suppressServiceRequests(suppressServiceRequests);
+        }
+
+        IFaultInjectionResult result = faultInjectionResultBuilder.build();
+
+        FaultInjectionCondition condition = new FaultInjectionConditionBuilder()
+            .operationType(operationType)
+            .connectionType(FaultInjectionConnectionType.DIRECT)
+            .build();
+
+        FaultInjectionRule rule = new FaultInjectionRuleBuilder("InjectedResponseDelay")
+            .condition(condition)
+            .result(result)
+            .build();
+
+        FaultInjectorProvider injectorProvider = (FaultInjectorProvider) container
+            .getOrConfigureFaultInjectorProvider(() -> new FaultInjectorProvider(container));
+
+        injectorProvider.configureFaultInjectionRules(Arrays.asList(rule)).block();
+
+        return rule;
+    }
+
+    private void executeAndValidate(
+        Supplier<Mono<CosmosItemResponse<ObjectNode>>> call,
+        String expectedId,
+        int expectedStatusCode,
+        boolean expectTrackingId) {
+
+        String activityId;
+        if (expectedStatusCode <= 299) {
+            CosmosItemResponse<ObjectNode> response = call.get().block();
+            assertThat(response.getStatusCode()).isEqualTo(expectedStatusCode);
+            assertThat(response.getItem().get("id").asText()).isEqualTo(expectedId);
+            assertThat(response.getItem().get("mypk").asText()).isEqualTo(expectedId);
+            activityId = response.getActivityId();
+            assertThat(activityId).isNotNull();
+            if (expectTrackingId) {
+                assertThat(response.getItem().get("_trackingId")).isNotNull();
```

Comment:
hmm, I do not think we should return the _trackingId in the response, it might cause deserialization issues if customer did not config ignore unknown?

Reply:
We need it in the document - will be documented as prerequisite when opting in
Alternative is to have one option to enable retries without the remapping based on tracking Id

Reply:
That is maybe worth considering - so having a config when enabling write retries -s specifying whether or not to use _trackingId - so, cx can decide whether they prefer more conflicts, pre-condition failures and want to deal with it themselves or whether we should do it in the document

Reply:
both won't work

Reply:
we can still keep the keep trackingId around until the very last step before we do the deserializing? 

and also if the item size already close to size limit, then they can not use this feature neither ?

Reply:
We can't remove it - customer could have added a _trackingId themselves
Either we take ownership of it or not IMO

Reply:
Changed trackingId usage to be optional